tests/libpod/query.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package libpod
+
+import (
+	"context"
+	"fmt"
+
+	"kubevirt.io/client-go/kubecli"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GetRunningPodByLabel(
+	virtCli kubecli.KubevirtClient,
+	label string,
+	labelType string,
+	namespace string,
+	node string) (*k8sv1.Pod, error) {
+
+	labelSelector := fmt.Sprintf("%s=%s", labelType, label)
+	var fieldSelector string
+	if node != "" {
+		fieldSelector = fmt.Sprintf("status.phase==%s,spec.nodeName==%s", k8sv1.PodRunning, node)
+	} else {
+		fieldSelector = fmt.Sprintf("status.phase==%s", k8sv1.PodRunning)
+	}
+	pods, err := virtCli.CoreV1().Pods(namespace).List(context.Background(),
+		metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: fieldSelector},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(pods.Items) == 0 {
+		return nil, fmt.Errorf("failed to find pod with the label %s", label)
+	}
+
+	var readyPod *k8sv1.Pod
+	for _, pod := range pods.Items {
+		// During migration there can be more than one pod
+		// In this case we want the target pod
+		if readyPod != nil && readyPod.CreationTimestamp.Before(&pod.CreationTimestamp) {
+			continue
+		}
+
+		readyPod = &pod
```

Comment:
My bad for the previous wrong suggestion.
Now that I am reading better, shouldn't this be reversed?
```suggestion
		if readyPod != nil && pod.CreationTimestamp.Before(&readyPod.CreationTimestamp) {
			continue
		}

		readyPod = &pod
```

We want to skip this `pod` if the "current" `readyPod` was created before it.
Or am I totally confused?
Thanks

Reply:
+1

Reply:
@xpivarc @fossedihelm Thanks folks!